### ISSUE TITLE
not exactly a dex-oracle issue

### ISSUE THREAD

**Issue by jc1122** [CalebFenton/dex-oracle#13]
Hello,

thank you for releasing this magnificent tool!
I am trying to look into the internal logic of this app:
https://www.apk4fun.com/apk/119729/

However, while trying to deobfuscate the mentioned app using dex-oracle, I received the advice clearly seen in the picture below:
![1](https://cloud.githubusercontent.com/assets/22999250/22399981/6075abc0-e5a9-11e6-98cd-7e1dc8c88132.JPG)

Using apktool v2.2.2: `apktool d mentionedApp.apk` I got the smali. The apk could be repacked without issues using `apktool b`.
As advised by dex-oracle, I replaced the `const-string` with `const-string/jumbo` in all smali files using:
`find ./ -type f exec sed -i -e 's/cont-string/const-string\/jumbo/g' {} \;`

However, the updated smali refused to rebuild to apk using `apktool b`. See picture below.
![2](https://cloud.githubusercontent.com/assets/22999250/22400048/34cb559a-e5ab-11e6-9c05-46602861bb2f.JPG)

Any leads?

Thank you in advance.

**Comment by CalebFenton** [comment]
Are you doing this on Windows?

It looks like your problem might be the "has already been interned". This means that dexlib (which apktool uses to build dex files) thinks it has already see the class `Lo/?;`. Since there's that `?` it makes me think that the file names for the classes somehow got messed up.

Try this: dump and rebuild with apktool without changing anything. If that doesn't work, it means the problem is in apktool.

Also try dumping with baksmali and rebuilding with smali. It might give you a better error message.

**Comment by jc1122** [comment]
Thanks for a really quick response!

Without doing any changes, the apk can be rebuilt without any problems.
The "has already been interned" error occurs only after changing `const-string`s to jumbo.
the only occurence of `const-string/jumbo` in smali or decompiled code with jadx is in bfl.java, line 357:
`        f11268 = new C2554if(26, "const-string", bfl.f10991, bfi.f10970);
        f11303 = new C2554if(27, "const-string/jumbo", bfl.f11003, bfi.f10970);`

Will try dumping with baksmali though.

**Comment by CalebFenton** [comment]
No problem.

It could be that dex-oracle is just stupidly looking for "const-string/jumbo" rather than the actual op `const-string/jumbo`, because who would ever have that as a string? Except this code looks like it's actually a string, and would fake that check out.

You want to change every opcode, not every string, so your regex might look something like `^\s*const-string ` (starts with zero or more spaces and ends with a string). Not sure why "const-string" or "const-string/jumbo" would be a string argument to a method, but you don't want to change either of them or you risk breaking the program.

**Comment by jc1122** [comment]
OK, I unzipped the app with 7-zip, then dumped classes.dex with baksmali (rebuilding the unmodified smali files with smali is successful without errors). Smali/baksmali v2.2b4.

Using notepad++, I searched in all the smali files for "const-string/jumbo", there was only one occurence and it was not an opcode but a string argument.
Then I replaced all occurences of "const-string" to "const-string/jumbo" (without replacing the string arg mentioned above) using notepad++ "find in files/replace in files" functionality. The dex file was rebuilt using `smali a .` to out.dex without errors. 

`dex-oracle -V out.dex` however exits giving exactly the same error like in the first message.

Any further suggestions?
